src/server/qgsrequesthandler.h

```diff
@@ -96,33 +96,26 @@ class QgsRequestHandler
     //! @note not available in Python bindings
     virtual void setGetCoverageResponse( QByteArray* ba ) = 0;
 
-    virtual void setDefaultHeaders() {}
-
     //! Set an HTTP header
     virtual void setHeader( const QString &name, const QString &value ) = 0;
 
     //! Remove an HTTP header
-    virtual int removeHeader( const QString &name ) = 0;
-
-    //! Delete all HTTP headers
-    virtual void clearHeaders() = 0;
+    virtual void removeHeader( const QString &name ) = 0;
 
     //! Append the bytestream to response body
     virtual void appendBody( const QByteArray &body ) = 0;
 
-    //! Clears the response body
-    virtual void clearBody() = 0;
```

Comment:
I was expecting this question :-)

Body and headers are related to each other, removing body without clearing the headers or removing all headers and keeping the whole body does not really make sense functionaly: I can't think of a use case where you would wipe out one without wiping the other - (As an example: all tests that where calling clear[Headers|Body] were calling both).

This is clearly related to what plugins can do with the content of the response: either they modify the content of the actual body and thus they don't really have to change the actual headers  or they build a completely new content and thus may eventually need to clear everything.